src/Agent.Worker/Handlers/Helpers/ProcessHandlerHelper.cs

```diff
@@ -119,49 +96,87 @@ private static int FindEnclosingIndex(string input, int targetIndex)
 
             return 0;
         }
-    }
 
+        public static (bool, Dictionary<string, object>) ValidateInputArguments(
+            string inputArgs,
+            Dictionary<string, string> environment,
+            IExecutionContext context)
+        {
+            var enableValidation = AgentKnobs.ProcessHandlerSecureArguments.GetValue(context).AsBoolean();
+            context.Debug($"Enable args validation: '{enableValidation}'");
+            var enableAudit = AgentKnobs.ProcessHandlerSecureArgumentsAudit.GetValue(context).AsBoolean();
+            context.Debug($"Enable args validation audit: '{enableAudit}'");
+            var enableTelemetry = AgentKnobs.ProcessHandlerTelemetry.GetValue(context).AsBoolean();
+            context.Debug($"Enable telemetry: '{enableTelemetry}'");
+
+            if (enableValidation || enableAudit || enableTelemetry)
+            {
+                context.Debug("Starting args env expansion");
+                var (expandedArgs, envExpandTelemetry) = ExpandCmdEnv(inputArgs, environment);
+                context.Debug($"Expanded args={expandedArgs}");
+
+                context.Debug("Starting args sanitization");
+                var (sanitizedArgs, sanitizeTelemetry) = CmdArgsSanitizer.SanitizeArguments(expandedArgs);
+
+                Dictionary<string, object> telemetry = null;
+                if (sanitizedArgs != inputArgs)
+                {
+                    if (enableTelemetry)
+                    {
+                        telemetry = envExpandTelemetry.ToDictionary();
+                        if (sanitizeTelemetry != null)
+                        {
+                            telemetry.AddRange(sanitizeTelemetry.ToDictionary());
+                        }
+                    }
+                    if (sanitizedArgs != expandedArgs)
```

Comment:
(not blocking) We might want to refactor so 'sanitizearguments' returns a value if escaping needed

Reply:
Sorry, what do you mean?
For now, `SanitizeArguments` returns sanitized value + telemetry and `ValidateInputArguments` returns `isValid` bool + telemetry.
What would we change?

Reply:
No issue, just a comment on the plumbing, not worth revisiting for now